backend/models.py

```diff
@@ -0,0 +1,57 @@
+
+# Create your models here.
+from django.contrib.auth.models import AbstractUser
+from django.db import models
+
+import uuid
+
+
+class User(AbstractUser):
+    # Using: username, password, first_name, last_name, email inherited from Abstractuser
+
+
+class Post(models.Model):
+    postId = models.UUIDField(
+        primary_key=True, default=uuid.uuid4, editable=False, unique=True)
+    author = models.ForeignKey(User, on_delete=models.CASCADE)
+    title = models.CharField(max_length=50)
+    timestamp = models.DateTimeField(auto_now_add=True)
+    description = models.CharField(max_length=450)
+    # default posts are public
+    isPublic = models.BooleanField(default=True)
+
+
+class PostAccess(models.Model):
+    author = models.ForeignKey(User, on_delete=models.CASCADE)
+    post = models.ForeignKey(Post, on_delete=models.CASCADE)
+    accessId = models.ForeignKey(User, on_delete=models.CASCADE)
+
+
+class Comments(models.Model):
+    commentId = models.UUIDField(
+        primary_key=True, uuid=uuid.uuid4, editable=False, unique=True)
+    content = models.CharField(max_length=250)
+    post = models.ForeignKey(Post, on_delete=models.CASCADE)
+    postedBy = models.ForeignKey(
+        User, on_delete=models.CASCADE, related_name="comment_postedBy")
+    postedTo = models.ForeignKey(
+        User, on_delete=models.CASCADE, related_name="comment_postedTo")
```

Comment:
I don't think we need postedTo since we only need to keep track of the postedBy user. If this is the case, we don't need related_name either since there won't be a clash. This is what i was thinking but i could be wrong.